src/servers/ws-server.ts

```diff
@@ -42,151 +55,189 @@ const getContext = async (
-        if (errors.length > 0) {
-          // returning `GraphQLError[]` sends an `ErrorMessage` and stops the subscription
-          return errors
-        }
-
-        // ready execution arguments
-        return args
-      },
-      onConnect: async (ctx) => {
-        baseLogger.debug("Connect", ctx)
-
-        const cookies = cookie.parse(ctx.extra.request.headers.cookie || "")
-        const kratosSessionCookie = cookies.ory_kratos_session
-
-        // TODO: integrate open telemetry
-        if (
-          typeof ctx.connectionParams?.Authorization !== "string" &&
-          !kratosSessionCookie
-        ) {
-          return true // anon connection ?
-        }
-
-        const context = await getContext(ctx)
-        if (typeof ctx.connectionParams?.Authorization === "string") {
-          authorizedContexts[ctx.connectionParams.Authorization] = context
-        }
-        if (kratosSessionCookie) {
-          authorizedContexts[kratosSessionCookie] = context
-        }
-
-        return true
-      },
-      context: async (ctx) => {
-        // TODO: integrate open telemetry
-        const cookies = cookie.parse(ctx.extra.request.headers.cookie || "")
-        const kratosSessionCookie = cookies.ory_kratos_session
-        // cookie auth
-        if (kratosSessionCookie) {
-          return authorizedContexts[kratosSessionCookie]
-        }
-        // bearer auth
-        if (typeof ctx.connectionParams?.Authorization === "string") {
-          return authorizedContexts[ctx.connectionParams?.Authorization]
-        }
-        // anon context
-        const context = await getContext(ctx)
-        return context
-      },
-      onNext: (ctx, msg, args, result) => {
-        baseLogger.debug("Next", { ctx, msg, args, result })
-      },
-      onError: (ctx, msg, errors) => {
-        baseLogger.debug("Error", { ctx, msg, errors })
-      },
-      onComplete: (ctx, msg) => {
-        baseLogger.debug("Complete", { ctx, msg })
-      },
+  wrapAsyncToRunInSpan({
+    namespace: "ws-server",
+    fnName: "useServer",
+    fn: async () => {
+      useServer(
+        {
+          onSubscribe: (_ctx, msg) => {
+            baseLogger.debug("Subscribe", { _ctx, msg })
+
+            // construct the execution arguments
+            const args = {
+              schema,
+              operationName: msg.payload.operationName,
+              document: parse(msg.payload.query),
+              variableValues: msg.payload.variables,
+            }
+
+            const operationAST = getOperationAST(args.document, args.operationName)
+            if (!operationAST) {
+              // returning `GraphQLError[]` sends an `ErrorMessage` and stops the subscription
+              recordExceptionInCurrentSpan({
+                error: new GraphQLError("Unable to identify operation"),
+                level: ErrorLevel.Warn,
+              })
+              return [new GraphQLError("Unable to identify operation")]
+            }
```

Comment:
@nicolasburtey ,

What do you recommend as the default error level for these?

Reply:
`warn` sounds good